### PR TITLE
Initial CI support

### PR DESCRIPTION
**Pull request by koppor**
This is a setup to solve https://github.com/reitzig/ltx2any/issues/116.

>     1. Run tests against
>     * Ruby versions 2.3, 2.4, 2.5;

**Update** Ruby 3.0, 3.1, 3.2, 3.3 included.

~~Simple execution done with these versions. Real LaTeXing test only Ruby 2.5, because even that is outdated now (see https://gitlab.alpinelinux.org/alpine/aports/-/issues/13555#note_218801)~~

>     * GNU/Linux and TeXlive;

**Update** Simple texlive installation on the runner (using https://github.com/zauguin/install-texlive)

~~Done via https://github.com/reitzig/texlive-docker~~

>     * macOS and MacTeX (?);

**Update** Uses TeXLive instead of MacTeX. Hope, this is OK?

~~Note done; one could follow the steps at https://github.com/actions/virtual-environments/issues/1143#issuecomment-652264388.~~
 
>     * Windows and MiKTeX/proTeXt (?).

**Udate** Done. But I have some ruby issues. See other comment below

~~With Docker hard to do https://github.com/actions/virtual-environments/issues/1143#issuecomment-651670345; even though MiKTeX offers a Docker image.~~

>     2. Build and deploy docs (if applicable).

Not done.

>     3. Build and deploy gem (if commit tagged).

Not done. Could be done with a Google Cloud Platform action: https://andrewm.codes/blog/automating-ruby-gem-releases-with-github-actions/

When this update is in place, the output on GitHub is as follows:

grafik

I don't know whether this works with `chew` (https://github.com/reitzig/ltx2any/tree/gemification), but I think, this can be adapted when the development is continued.

Side comment: The motivation to look into this is that installing ruby on a Docker image is less a hazzle than ensuring that `makeglossaries` is called at the correct place in other build tools. [cluttex](https://github.com/minoki/cluttex) was my choice after `latexmk`, because it is written in lua, which is available in LaTeX installations. However, it does trigger `bibtex` when a `.bib` file is updated. -- Maybe, a stand-alone binary can be generated in the future so that one does not need to play around with ruby versions.

### CODE REVIEW

**Comment by koppor** [comment]
I updated the PR to include macOS, Windows, and some more ruby versions.

Maybe, the number of ruby versions can be reduced?

I also have issues with Ruby on Windows:

```
Run ruby ltx2any.rb test/test.tex
[ltx2any] Copying files to tmp ... Done
pdflatex: security risk: running with elevated privileges
[ltx2any] PdfLaTeX(1) running ... Error
[ltx2any] Error: uninitialized constant PdfLaTeX::TexLogParser
          See test.err for details.
```

**Review by reitzig — CHANGES_REQUESTED** [review]
Let's drop support for Ruby <3.1.

All of 2.x are EOL, and 3.0 is about to be: https://endoflife.date/ruby

**Comment by reitzig** [comment]
Can't seem to update the PR myself; will add suggestions.

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+jobs:
+  test-pdflatex-linux-windows:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        ruby: [ '2.3', '2.4', '2.5', '3.0', '3.1', '3.2', '3.3' ]
```
Comment (reitzig) [inline .github/workflows/check.yml:16]:
```suggestion
        ruby: [ '3.1', '3.2', '3.3' ]
```

File: .github/workflows/check.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+          packages: >
+            latex-bin scheme-basic
+      - name: ltx2any
+        run: |
+          ruby ltx2any.rb test/test.tex
+  test-pdflatex-win:
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: [ '2.3', '2.4', '2.5', '3.0', '3.1', '3.2', '3.3' ]
```
Comment (reitzig) [inline .github/workflows/check.yml:41]:
```suggestion
        ruby: [ '3.1', '3.2', '3.3' ]
```

File: README.md [hunk 3]
```diff
@@ -32,7 +32,7 @@ bug-free or reliable. Take care!
 
 For using ltx2any without any bells and whistles, you should have
 
- * Ruby 2.3.0 or higher and
+ * Ruby 2.3, 2.4., or 2.5 and
```
Comment (reitzig) [inline README.md:35]:
```suggestion
 * Ruby 3.1 or higher, and
```

**Review by reitzig — APPROVED** [review]

**Comment by koppor** [comment]
Note that the Windows MiKTeX install cannot be performance increased, because Caching for `choco install` is still an open issue: https://github.com/chocolatey/choco/issues/2134

**Comment by koppor** [comment]
Any improvements to the MiKTeX workflow should be contributed back to https://stackoverflow.com/a/76953318/873282.

**Pull request merged**